core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -226,50 +249,32 @@ private[spark] class ExecutorAllocationManager(
    * If the cap on the number of executors is reached, give up and reset the
    * number of executors to add next round instead of continuing to double it.
    * Return the number actually requested.
+   *
+   * @param maxNumNeededExecutors the maximum number of executors all currently running or pending
+   *                              tasks could fill
    */
-  private def addExecutors(): Int = synchronized {
+  private def addExecutors(maxNumNeededExecutors: Int): Int = synchronized {
```

Comment:
i think addExecutors donot need to return Int value. It is unnecessary because the returned value have not be used.